src/DiskIO/IpcIo/IpcIoFile.cc

```diff
@@ -447,23 +444,57 @@ IpcIoFile::canWait() const
     return false; // do not want to wait that long
 }
 
-/// called when coordinator responds to worker open request
 void
-IpcIoFile::HandleOpenResponse(const Ipc::StrandMessage &response)
+IpcIoFile::HandleStrandReadyResponse(const Ipc::StrandReady &response)
 {
-    debugs(47, 7, HERE << "coordinator response to open request");
-    for (IpcIoFileList::iterator i = WaitingForOpen.begin();
-            i != WaitingForOpen.end(); ++i) {
-        if (response.strand.tag == (*i)->dbName) {
-            (*i)->openCompleted(&response);
-            WaitingForOpen.erase(i);
-            return;
-        }
+    debugs(47, 7, "disker" << response.strand.kidId <<  " " << response.strand.tag << " is ready");
+    if (const auto file = StopWaiting(response.strand))
+        file->openCompleted(&response);
+    else
+        debugs(47, 4, "premature or late disker response");
```

Comment:
"Premature" kind of implies that we have sent a request and received a response earlier than expected. AFAICT, this is the case where this kid did not send a request yet.

```suggestion
        debugs(47, 4, "foreign or late disker response");
```

or

```suggestion
        debugs(47, 4, "unsolicited or late disker response");
```

However, if that situation is no longer possible because our caller will protect us from foreign/unsolicited messages, then we can simplify:

```suggestion
        debugs(47, 4, "late disker response");
```